src/DDTrace/Integrations/AMQP/AMQPIntegration.php

```diff
@@ -167,6 +167,63 @@ function ($This) use ($integration) {
             ]
         );
 
+        trace_method(
+            "PhpAmqpLib\Channel\AMQPChannel",
+            "batch_basic_publish",
+            [
+                'prehook' => function (SpanData $span, $args) use ($integration) {
+                    /** @var AMQPMessage $message */
+                    $message = $args[0];
+                    if (!is_null($message)) {
+                        $integration->injectContext($message);
+                    }
+                },
+                'posthook' => function (SpanData $span, $args, $exception) use ($integration) {
+                    /** @var AMQPMessage $message */
+                    $message = $args[0];
+                    /** @var string $exchange */
+                    $exchange = $args[1];
+                    /** @var string $routing_key */
+                    $routingKey = $args[2] ?? '';
+
+                    $exchangeDisplayName = $integration->formatExchangeName($exchange);
+                    $routingKeyDisplayName = $integration->formatRoutingKey($routingKey);
+
+                    $integration->setGenericTags(
+                        $span,
+                        'batch_basic.add',
+                        'producer',
+                        "$exchangeDisplayName -> $routingKeyDisplayName",
+                        $exception
+                    );
+                    $span->meta[Tag::MQ_OPERATION] = 'batch';
```

Comment:
For semantical and consistency reasons between traces, the `messaging.operation` tag must be one of:
- `send`
- `receive`
- `process`

You are not supposed to know this :) In this case, `send` should be used instead of `batch`.

The tag should be changed in the tests as well :+1:

Reply:
Should it be `process`, then? I mean, that operation isn't sending nor receiving anything :)

Reply:
Both `receive` and `process` are reserved for messages consumer/server, so we cannot use either in this specific case :( 

As you correctly pointed out, this operation isn't a producer _per se_ because it doesn't send anything... (Let's keep it as a producer though, as it makes sense with the [OTel spec](https://github.com/open-telemetry/opentelemetry-specification/blob/9fa7c656b26647b27e485a6af7e38dc716eba98a/specification/trace/api.md#spankind)) There are two possibilities, I believe:
1. We use `send`, but it's misleading
2. We don't put a `messaging.operation` tag

I believe the second option makes the most sense for this operation, as this tag is not mandatory. If it doesn't make sense, then let's keep it simple imo